src/Microsoft.AspNetCore.Razor.Tools/ServerCommand.cs

```diff
@@ -89,5 +105,40 @@ protected virtual void ExecuteServerCore(ConnectionHost host, CompilerHost compi
             var dispatcher = RequestDispatcher.Create(host, compilerHost, cancellationToken, eventBus, keepAlive);
             dispatcher.Run();
         }
+
+        internal FileStream WritePidFile()
+        {
+            // To make all the running rzc servers more discoverable, We want to write the process Id and pipe name to a file.
+            // The file contents will be in the following format,
+            //
+            // <PID>
+            // rzc
+            // <pipename>
+
+            const int DefaultBufferSize = 4096;
+            var processId = Process.GetCurrentProcess().Id;
+            var fileName = $"rzc-{processId}";
+
+            var path = Environment.GetEnvironmentVariable("DOTNET_BUILD_PIDFILE_DIRECTORY");
+            if (string.IsNullOrEmpty(path))
+            {
+                var homeEnvVariable = PlatformInformation.IsWindows ? "USERPROFILE" : "HOME";
+                path = Path.Combine(Environment.GetEnvironmentVariable(homeEnvVariable), ".dotnet", "pids", "build");
```

Comment:
We should probably guard against `Environment.GetEnvironmentVariable(homeEnvVariable)` returning null (`dotnet` has an outstanding issue in CI environments that unset `HOME`, for example).  I think the correct behavior would be to skip writing the file.